src/shared/XLinkDevice.c

```diff
@@ -343,9 +343,9 @@ XLinkError_t XLinkResetRemoteTimeout(linkId_t id, int timeoutMs)
     XLinkError_t ret = DispatcherWaitEventCompleteTimeout(&link->deviceHandle, absTimeout);
 
     if(ret != X_LINK_SUCCESS){
-        // Close remote causes to close any links which unblocks the previous events
-        // It cleans the rest of dispatcher properly
-        DispatcherReset(&link->deviceHandle);
+        // Closing device link unblocks any blocked events
+        // Afterwards the dispatcher can properly cleanup in its own thread
+        DispatcherDeviceFdDown(&link->deviceHandle);
```

Comment:
Should this same if/DisDeviceFdDown() block of code also in be the non-timeout version `XLinkResetRemote()`?

Reply:
There is "no specific time" at which the device communication should be closed "forcefully". Thats why its only used in the timeout case, where there is "a specified amount of time" until the operation should finish (timeout might not be the best name in that case but yeah)